### PR TITLE
Bundle `features` package through rollup

### PR DESCRIPTION
**Pull request by thiva-k**
### Purpose

This PR lays the foundation for bundling `features` package into a self-contained JavaScript ESM package that can be published to npm and reused in other applications.

#### Changes made:
- Created a `public-api.ts` for each feature to expose through the package.
- Created a stand-alone `tsconfig.json` for features package without extending `tsconfig.base.json`.
- Added `main` and `types` fields in package.json of  `react-components`, `forms`, and `dynamic-forms` modules.
- Configured `rollup.config.cjs` to bundle `features` .
 - - Bundled as separate JS files preserving the initial package structure rather than a single bundled output file.
 - - Configured Rollup plugins to handle various file types and type declarations.
 - - Configured to generate a separate `package.json` with 'exports' fields pointing to public-api.js of each feature through rollup for the bundled output in `dist/esm`.  (To continue `identity-apps` without any changes for now due to building and bundling of `features` package) and enable publishing the bundled files manually through this generated `package.json` if needed.

### Related Issues
- product-is: https://github.com/wso2/product-is/issues/20119

### Related PRs
- identity-apps: https://github.com/wso2/identity-apps/pull/5849

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.api-resources.v1/public-api.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export { default as APIResourcesEditPage } from "./pages/api-resource-edit";
+export { default as APIResourcesPage } from "./pages/api-resources";
+
+
```
Comment (brionmario) [inline features/admin.api-resources.v1/public-api.ts:22]: Let's remove the additional line break.
Check other files as well.

Comment (thiva-k) [reply features/admin.api-resources.v1/public-api.ts]: Removed

File: features/admin.core.v1/public-api.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export * from "./utils";
+export * from "./api";
+
+/**
+ * Export the components.
+ */
+export * from "./components/quick-start-tab";
+
+export * from "./components/modals/help-panel-modal";
+// export * from "./components/modals/tier-limit-error-modal";
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply features/admin.core.v1/public-api.ts]: Removed

**Review by JayaShakthi97 — APPROVED** [review]

File: features/package.json [hunk 3]
```diff
@@ -73,6 +73,7 @@
         "redux-mock-store": "^1.5.4",
         "redux-thunk": "^2.3.0",
         "regenerator-runtime": "^0.13.9",
+        "rollup": "^4.17.2",
```
[the remark this answers is not in the document]
Comment (thiva-k) [reply features/package.json]: Moved to dev dependency

**Review by brionmario — APPROVED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -197,7 +197,8 @@ module.exports = {
             "files": "*.json",
             "parser": "jsonc-eslint-parser",
             "rules": {
-                "header/header": "off"
+                "header/header": "off",
+                "max-len": "off"
```
Comment (brionmario) [inline .eslintrc.js:201]:
```suggestion
                "max-len": "off",
                "semi": "off"
```

**Review by brionmario — APPROVED** [review]

File: features/admin.certificates.v1/index.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```
Comment (pavinduLakshan) [inline features/admin.certificates.v1/index.ts:2]:
```suggestion
 * Copyright (c) 2020-2024, WSO2 LLC. (https://www.wso2.com).
```

File: features/admin.connections.v1/index.ts [hunk 6]
```diff
@@ -69,6 +69,7 @@ export * from "./models/endpoints";
  */
 export * from "./pages/connection-templates";
 export * from "./pages/connections";
+export * from "./pages/connection-edit";
```
Comment (pavinduLakshan) [inline features/admin.connections.v1/index.ts:75]:
```suggestion

```

File: features/admin.extensions.v1/public-api.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ * under the License.
+ */
+
+export * from "./extensions-manager";
+export * from "./models";
+export * from "./configs";
+export { useGetAllFeatures } from "./components/feature-gate/api/feature-gate";
+export { featureGateConfig } from "./configs/feature-gate";
+export { CONSUMER_USERSTORE } from "./components/administrators/constants/users";
+export {  TenantTier } from "./components/subscription/models/subscription";
```
Comment (pavinduLakshan) [inline features/admin.extensions.v1/public-api.ts:25]:
```suggestion
export { TenantTier } from "./components/subscription/models/subscription";
```

File: features/admin.remote-repository-configuration.v1/index.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```
Comment (pavinduLakshan) [inline features/admin.remote-repository-configuration.v1/index.ts:2]:
```suggestion
 * Copyright (c) 2020-2024, WSO2 LLC. (https://www.wso2.com).
```

File: features/admin.tenants.v1/configs/index.ts [hunk 9]
```diff
@@ -16,5 +16,6 @@
  * under the License.
  */
 
-export * from "./endpoints";
 export * from "./ui";
+export * from "./endpoints";
+
```
Comment (pavinduLakshan) [inline features/admin.tenants.v1/configs/index.ts:21]:
```suggestion
```

**Review by pavinduLakshan — APPROVED** [review]

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**